spyder/plugins/application/container.py

```diff
@@ -106,15 +106,21 @@ def setup(self):
 
         # Attributes
         self.dialog_manager = DialogManager()
-        self.application_update_status = None
-        if is_pynsist() or running_in_mac_app():
-            self.application_update_status = ApplicationUpdateStatus(
-                parent=self)
-            (self.application_update_status.sig_check_for_updates_requested
-             .connect(self.check_updates))
-            (self.application_update_status.sig_install_on_close_requested
-                 .connect(self.set_installer_path))
-            self.application_update_status.set_no_status()
+
+        self.application_update_status = ApplicationUpdateStatus(
+            parent=self
+        )
+
+        # Users can only use this widget in our apps.
+        if not is_pynsist() and not running_in_mac_app():
+            self.application_update_status.hide()
```

Comment:
I think this needs to be ported to `master`, I mean, added to the right location, right?

Reply:
Well, I supposed that the `ApplicationUpdateStatus` status bar widget would be visible for all Spyder installs. Is there a reason why would want to hide it for standard installations of Spyder >=6.0?

Reply:
If we want it visible, then there is nothing to do in master. If we want it hidden, then we would need to port it to the correct location in master. I can do this with a separate PR if you like.

Reply:
> Is there a reason why we would want to hide it for standard installations of Spyder >=6.0 and visible only for conda-based installs?

It doesn't need to be available for development (i.e. when `DEV == True`) because it's unnecessary.

But also, when is the widget shown? Only when an update is available? Or all the time? The logic in `master` seems to imply that it's visible all the time, which makes little sense to me because it'll waste precious space in the status bar when no update is available.

> If we want it hidden, then we would need to port it to the correct location in master. I can do this with a separate PR.

Ok, then I'll merge this PR to master without any changes so that you can address this in a separate one.